lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -1085,7 +1102,12 @@ struct ConvertGenericToVnniMatmulLikeOp
     int64_t kPos = 1;
     if (hasBatch)
       kPos++;
-    int64_t k = cast<ShapedType>(bufferA.getType()).getShape()[kPos];
+    // Take the whole reduction dim size. Account for the VNNI factor that
+    // splits the K dim in the shape.
+    std::optional<int64_t> vnniFactor =
+        vnni::utils::getVnniBlockingFactor(bufferB.getType());
+    int64_t k =
+        cast<ShapedType>(bufferA.getType()).getShape()[kPos] * *vnniFactor;
```

Comment:
You haven't checked that the `optional` has a value. If it does not, this will crash.

Reply:
The VNNI layout is ensured by the earlier checks. Added comment plus assert just in case.